6MI0800226/week2.cpp

```diff
@@ -0,0 +1,160 @@
+#include <iostream>
+#include <fstream>
+#include <assert.h>
+ 
+const int BUFF_SIZE = 1024;
+void saveResultsInFile(int num1, int num2, const char location[BUFF_SIZE])
+{
+    std::ofstream file(location);
+ 
+    if (!file.is_open())
+        return;
+ 
+    file << (num1 + num2) << std::endl << (num1 * num2);
+}
+int findDiffInFile(const char location[BUFF_SIZE])
+{
+    int num1, num2;
+    std::ifstream file(location);
+ 
+    if (!file.is_open())
+        return -1;
+ 
+    file >> num1 >> num2;
+    return num1 - num2;
+}
+ 
+int sizeOfFile(const char location[BUFF_SIZE])
+{
+    int size = 0;
+    std::ifstream file(location);
+ 
+    if (!file.is_open())
+        return -1;
+ 
+    while (!file.eof())
+    {
+        char buffM[BUFF_SIZE];
+        char* buff = buffM;
+        file.getline(buffM, BUFF_SIZE);
+        while (*buff)
+        {
+            if (*buff < 'a' || *buff > 'd') size++;
+            buff++;
+        }
+    }
+    file.close();
+    return size;
+}
+ 
+struct pair
+{
+    int x, y;
+};
+pair createPair(int x, int y)
+{
+    pair p;
+    p.x = x;
+    p.y = y;
+    return p;
+}
+void initPair(pair& toInit, int x, int y)
+{
+    toInit.x = x;
+    toInit.y = y;
+}
+pair readPairFromFile(std::ifstream& inFile)
+{
+    return pair{ inFile.get(), inFile.get() };
```

Comment:
This way you read a single character, not a number. This will not work correctly :))